app/helpers.py

```diff
@@ -55,11 +65,13 @@ def cloning_document(document_id, auth_viewers, parent_id, process_id):
             else [auth_viewers]
         )
         document = get_document_object(document_id)
-        doc_name = document["document_name"]
-        if has_tilde_characters(doc_name):
-            document_name = doc_name
-        else:
-            document_name = "~" + doc_name + "~"
+        for viewer in viewers:
+            doc_name = document["document_name"]
+            if has_tilde_characters(doc_name):
+                document_name = doc_name.replace('~', '')
+            else:
+                rand = randomize(10)
+                document_name = doc_name + "_clone_" + viewer + f"_{rand}"
```

Comment:
Keep the document very simple. doc_name + viewer, no need for _clone_ and random string